### ISSUE TITLE
Unable to create keyvault behind proxy as net.dial does not use proxy

### ISSUE THREAD

**Issue by oddperfect** [terraform-providers/terraform-provider-azurerm#604]
### Terraform Version
Terraform v0.11.1
+ provider.azurerm v0.3.3

### Affected Resource(s)
- azurerm_key_vault

### Terraform Configuration Files
any keyvault config will do 

### Debug Output
module.keyvaults.azurerm_key_vault.keyvault: Creating...
this statement after creating mutiple times until 2m30: module.keyvaults.azurerm_key_vault.keyvault: Still creating... 
Then
1 error(s) occurred:

* module.keyvaults.azurerm_key_vault.keyvault: 1 error(s) occurred:

* azurerm_key_vault.keyvault: dial tcp *******:443: getsockopt: operation timed out

### Panic Output
None

### Expected Behavior
Keyvault successfully created

### Actual Behavior
Keyvault is created in azure but terraform is unable to reach the url for the keyvault when behind proxy

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Be behind a proxy without direct access to azure.
2. `terraform apply`

### Important Factoids
On OSX.
Relative environment variables set that do not seem to affect it: 
- http_proxy
- https_proxy
- HTTP_PROXY
- HTTPS_PROXY
- all_proxy
- ALL_PROXY

Using wireshark I can see that only direct connections are made to the keyvault. Does not go thru proxy. 

### References
None

**Comment by donestefan** [comment]
Same Problem here in Windows 7
```
* azurerm_key_vault.my-pw-test: Error waiting for Key Vault "my-pw-test" (Resource Group "test-rg") to become available: Error connecting to "my-pw-test.vault.azure.net:443": dial tcp x.x.x.x:443: connectex: No connection could be made because the target machine actively refused it.
```
Using a network sniffer I confirmed that all terraform connections go through the proxy. 
Only the connections testing the successfull creation ignore the proxy and go direct (and will usually be rejected by firewall).

**Comment by lstolyarov** [comment]
+1 seeing the same error behind a proxy.

@tombuildsstuff any ideas how this can be solved?

**Comment by jondkent** [comment]
+1 

Would 'thumbs up' this, but for some reason thats not available here.

**Comment by katbyte** [comment]
@jondkent, @lstolyarov, @donestefan, @oddperfect:

Just a heads up this has been included in the 1.7 release available as of this afternoon, hope it resolves this for everyone!

**Comment by oddperfect** [comment]
Thank you! Just tried it out and it worked perfectly. This had been blocking me from updating past version 0.2.2 so I really appreciate the work.
